src/template_partials/templatetags/partials.py

```diff
@@ -155,21 +155,27 @@ def __init__(self, parent_dict, lookup_key):
         self.lookup_key = lookup_key
 
     def __getitem__(self, key):
-        return self.parent_dict[self.lookup_key][key]
-
-
-class OriginPartialContentsWrapper:
-    """
-    Wrap parser.origin to allow deferred access to partial_contents.
-    """
-
-    def __init__(self, origin):
-        self.origin = origin
-
-    def __getitem__(self, key):
-        if not hasattr(self.origin, "partial_contents"):
-            self.origin.partial_contents = {}
-        return self.origin.partial_contents[key]
+        try:
+            # Try Django 5.1+ dict-based storage first
+            partials_content = self.parent_dict[self.lookup_key]
+        except KeyError:
+            raise template.TemplateSyntaxError(
+                f"You have not defined any partial. You are trying to access '{key}' partial"
+            )
+        except TypeError:
+            # Fall back to pre-Django 5.1 object-based storage
+            if not hasattr(self.parent_dict, self.lookup_key):
+                raise template.TemplateSyntaxError(
+                    f"You have not defined any partial. You are trying to access '{key}' partial"
+                )
+            partials_content = getattr(self.parent_dict, self.lookup_key)
```

Comment:
Again here, instead of using `if not hasattr...` simply `try` the `getattr` call and catch the `AttributeError` in the exception block.

Reply:
oh, missed that one : ) . Updated